pilot/pkg/xds/endpoint_builder.go

```diff
@@ -315,9 +323,11 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 				}
 				localityEpMap[ep.Locality.Label] = locLbEps
 			}
-			if ep.EnvoyEndpoint == nil {
-				ep.EnvoyEndpoint = buildEnvoyLbEndpoint(ep)
+			tlsMode := ep.TLSMode
+			if b.mtlsChecker != nil && b.mtlsChecker.mtlsDisabledByPeerAuthentication(ep) {
+				tlsMode = ""
```

Comment:
Does this also make sense to do with the client settings from DR?

Reply:
If we do have client DR we shouldn't use auto mtls so it probably doesn't matter?

Reply:
you mean when the client side DR disable the mTLS? I think in that case the cds.go code would already configure the cluster as plaintext, no `transport_socket_matches` at all.